plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/StatsMetricPublisher.java

```diff
@@ -41,15 +93,26 @@ public void close() {}
     public MetricPublisher getObjectMetricPublisher = new MetricPublisher() {
         @Override
         public void publish(MetricCollection metricCollection) {
-            stats.getCount.addAndGet(
-                metricCollection.children()
-                    .stream()
-                    .filter(
-                        metricRecords -> metricRecords.name().equals("ApiCallAttempt")
-                            && !metricRecords.metricValues(HttpMetric.HTTP_STATUS_CODE).isEmpty()
-                    )
-                    .count()
-            );
+            for (MetricRecord<?> metricRecord : metricCollection) {
+                switch (metricRecord.metric().name()) {
+                    case "ApiCallDuration":
+                        extendedStats.get(BlobStore.Metric.REQUEST_LATENCY).getMetrics.addAndGet(
+                            ((Duration) metricRecord.value()).toMillis()
+                        );
+                        break;
+                    case "RetryCount":
+                        extendedStats.get(BlobStore.Metric.RETRY_COUNT).getMetrics.addAndGet(((Integer) metricRecord.value()));
+                        break;
+                    case "ApiCallSuccessful":
+                        if ((Boolean) metricRecord.value()) {
+                            extendedStats.get(BlobStore.Metric.REQUEST_SUCCESS).getMetrics.addAndGet(1);
```

Comment:
Regarding the format of the output , the current format is 

```
         "request_retry_count_total" : {
            "ListObjects" : 0,
            "PutMultipartObject" : 0,
            "DeleteObjects" : 0,
            "PutObject" : 0,
            "GetObject" : 0
          },
          "request_time_in_millis" : {
            "ListObjects" : 7822,
            "PutMultipartObject" : 0,
            "DeleteObjects" : 2929,
            "PutObject" : 30366,
            "GetObject" : 20428
          }
```

Should we go with 

```
         "ListObjects " : {
            "request_retry_count_total" : 0,
            "request_time_in_millis" : 0,
          },
          "PutMultipartObject" : {
            "request_retry_count_total" : 0,
            "request_time_in_millis" : 0,
          }
```

The above seems more intuitive for the user , as they would want to see how a specific API is behaving .

Reply:
We need to see in the context of the parent blob which is repository and generic enough for any cloud vendor. We can't place `ListObjects` etc above since it's specific to S3 and we want the format to be largely consistent across vendors and hence the structure.